sklearn/model_selection/_validation.py

```diff
@@ -992,10 +1136,60 @@ def validation_curve(estimator, X, y, param_name, param_range, groups=None,
         parameters={param_name: v}, fit_params=None, return_train_score=True)
         # NOTE do not change order of iteration to allow one time cv splitters
         for train, test in cv.split(X, y, groups) for v in param_range)
+    out = list(zip(*out))
 
-    out = np.asarray(out)
     n_params = len(param_range)
-    n_cv_folds = out.shape[0] // n_params
-    out = out.reshape(n_cv_folds, n_params, 2).transpose((2, 1, 0))
+    n_splits = cv.get_n_splits(X, y)
+
+    out[0] = (np.array(out[0]).reshape(n_splits, n_params)).T
+    out[1] = (np.array(out[1]).reshape(n_splits, n_params)).T
 
     return out[0], out[1]
+
+
+def _aggregate_score_dicts(scores, shape=None, transpose=False):
+    """Aggregate the list of dict to dict of np ndarray
```

Comment:
It doesn't look like either shape or transpose is currently used. Do we need them?